packages/key-management/src/InMemoryKeyAgent.ts

```diff
@@ -55,11 +57,15 @@ export class InMemoryKeyAgent extends KeyAgentBase implements KeyAgent {
     this.#getPassphrase = getPassphrase;
   }
 
-  async signBlob({ index, role: type }: AccountKeyDerivationPath, blob: HexBlob): Promise<SignBlobResult> {
+  async signBlob(
+    { index, role: type, purpose = KeyPurpose.STANDARD }: AccountKeyDerivationPath,
+    blob: HexBlob
+  ): Promise<SignBlobResult> {
     const rootPrivateKey = await this.#decryptRootPrivateKey();
     const accountKey = await deriveAccountPrivateKey({
       accountIndex: this.accountIndex,
       bip32Ed25519: this.bip32Ed25519,
+      purpose,
```

Comment:
```suggestion
      purpose: this.purpose,
```

Reply:
Where is `this.purpose` defined? `KeyAgentBase` does not have a class variable `purpose`. Should I add a `purpose` property to [KeyAgent](https://github.com/input-output-hk/cardano-js-sdk/blob/82449e7dd25a6b291c855110d1487d1d4d3e3a85/packages/key-management/src/types.ts#L191)?